test/lib/render-helpers.js

```diff
@@ -19,8 +19,10 @@ import { setupInitialStore, connectToBackground } from '../../ui';
 import Root from '../../ui/pages';
 
 // Mock MetaMetrics context for tests
-const createMockTrackEvent = () => {
-  const mockTrackEvent = () => Promise.resolve();
+const createMockTrackEvent = (
+  getMockTrackEvent = () => () => Promise.resolve(),
+) => {
+  const mockTrackEvent = getMockTrackEvent();
```

Comment:
These changes were necessary because even though we create a mock `trackEvent` function in tests it was not obvious to me how to access that in tests so we could verify that Segment events were created correctly. These changes allow us to do that (see how they are used in `network-connection-banner.test.tsx`).